src/Stryker.Core/Stryker.Core/Initialisation/Buildalyzer/IAnalyzerResultExtensions.cs

```diff
@@ -183,7 +183,16 @@ internal static string GetAssemblyOriginatorKeyFile(this IAnalyzerResult analyze
         var warningsNotAsErrorsString = analyzerResult.GetPropertyOrDefault("WarningsNotAsErrors");
         var warningsNotAsErrors = warningsNotAsErrorsString is not null ? warningsNotAsErrorsString.Split(";").Where(x => !string.IsNullOrWhiteSpace(x)).ToDictionary(x => x, x => ReportDiagnostic.Warn) : new Dictionary<string, ReportDiagnostic>();
 
-        var diagnosticOptions = noWarn.Concat(warningsAsErrors).Concat(warningsNotAsErrors);
+        // merge settings,
```

Comment:
I think we should at the very least document this behavior, but preferably find the exact logic used by msbuild which apparently accepts this configuration and copy that. With this implementation it seems like we're making assumptions about the severity of the diagnostic.

My personal preference when we can't find what msbuild does would be to treat warningsAsErrors as more imporant than the reverse, so we should process that last.

Reply:
agree it should be documented. My ideal is that it followed what actually happens, but I fail to find any documentation about this (it seems to be undocumented).
The problem with prioritizing warnaserror above the errors is that the project may not compile at all without any mutations. So I have chosen a more tolerant approach

Reply:
Actually never mind, this is safer because the user can always decide that they had intended for the mutant to be killed by the error and they would then know they made a configuration mistake instead of the mutant being silently killed when they did not intend it.